# Exercise 5.29

In [1]:
import pathlib
import numpy as np
import pandas as pd
import numpy.linalg as la
from scipy import stats

In [2]:
def load_data() -> pd.DataFrame:
    # Automobile nominal thickness data.
    fpth = pathlib.Path(r'..\..\data\Table5.14.xlsx')
    return pd.read_excel(fpth)

In [3]:
# Keep only the first 30 observations.
thick_df = load_data().iloc[:30, :]
thick_df.head(3)

x1    x2    x3    x4    x5    x6
0 -0.12  0.36  0.40  0.25  1.37 -0.13
1 -0.60 -0.35  0.04 -0.28 -0.25 -0.15
2 -0.13  0.05  0.84  0.61  1.45  0.25

In [4]:
X = thick_df.to_numpy()
n, p = X.shape

In [5]:
xbar = np.mean(X, axis=0).reshape(p, 1)
S = np.cov(X, rowvar=False)

In [6]:
alpha = 0.05
mu0 = np.array([[0], [0], [0], [0], [0], [0]])
f_crit = (((n-1)*p)/(n-p))* stats.f(dfn=p, dfd=n-p).ppf(q=1-alpha)
f_crit

18.1843689698186

$$
H_{0}: \bm{\mu} = \textbf{0} = \bm{\mu}_{0}
\hspace{0.2cm}
\text{ versus }
\hspace{0.2cm}
H_{1}: \bm{\mu} \ne \textbf{0}
$$

Compute
$$
T^{2}
=
n
{(\bar{\textbf{x}} - \bm{\mu}_{0})}^{\prime}
\textbf{S}^{-1}
(\bar{\textbf{x}} - \bm{\mu}_{0})
$$

In [7]:
T2 = n*(xbar - mu0).T @ la.inv(S) @ (xbar - mu0)
T2.item()

374.7226876093117

In [8]:
T2 < f_crit

array([[False]])

Our $T^{2}$ value of 374.72 is larger than our critical valuee of 18.18, so we'd reject the null hypothesis that $\bm{\mu} = \textbf{0}$ at the 0.05-level.